projects/batfish-client/src/org/batfish/client/Client.java

```diff
@@ -315,6 +323,151 @@ private boolean answer(String questionTemplateName, String paramsLine,
+    * Validate the json-encoded {@code value} matches the valid setting
+    *  specified by {@code variable}.
+    *  <p>If {@code value} does not match the valid setting,
+    *  {@link BatfishException BatfishException} will be thrown.</p>
+    */
+   static void validate(JsonNode value, Variable variable)
+         throws BatfishException {
+      if (variable.getType().getStringType() &&
+            !variable.getType().equals(Variable.Type.JSON_PATH)) {
+         if (!value.isTextual()) {
+            throw new BatfishException("The parameter value type: " + value
+                  .getNodeType() + " doesn't match the type required for " +
+                  "variable type: " + variable.getType());
+         }
+      }
+      switch (variable.getType()) {
+         case BOOLEAN:
+            if (!value.isBoolean()) {
+               throw new BatfishException("Expected a boolean for " +
+                     "variable type: " + variable.getType());
+            }
+            break;
+         case COMPARATOR:
+            String textContent = value.textValue();
+            if (!(textContent.equals(">") || textContent.equals(">=") ||
+                  textContent.equals("==") || textContent.equals("!=") ||
+                  textContent.equals("<=") || textContent.equals("<"))) {
+               throw new BatfishException("Expected a comparator for " +
+                     "variable type: " + variable.getType());
+            }
+            break;
+         case INTEGER:
+            if (!value.isInt()) {
+               throw new BatfishException("Expected an Integer for " +
+                     "variable type: " + variable.getType());
+            }
+            break;
+         case IP:
+            //TODO: Need to double check isInetAddress()
+            if (!InetAddresses.isInetAddress(value.textValue())) {
+               throw new BatfishException("Not a valid ip address for " +
+                     "variable type: " + variable.getType());
+            }
+            break;
+         case IP_WILDCARD:
+            //TODO: need ip_WildCard validation check
+            break;
+         case JAVA_REGEX:
+            try {
+               Pattern.compile(value.textValue());
+            } catch (PatternSyntaxException e) {
+               throw new BatfishException("Expected a Java regular expression" +
+                     " for variable type: " + variable.getType());
+            }
+            break;
+         case JSON_PATH_REGEX:
+            validateJsonPathRegex(value.textValue());
+            break;
+         case PREFIX:
+            //TODO: need prefix validation check
+            break;
+         case PREFIX_RANGE:
+            //TODO: need prefix range validation check
+            break;
+         case STRING:
+            break;
+         case SUBRANGE:
+            //TODO: need prefix range validation check
+            break;
+         case JSON_PATH:
+            validateJsonPath(value);
+            break;
```

Comment:
in all of the above, please make sure the invalid value is printed as well as the name of the parameter that is invalid.